packages/Poll/index.js

```diff
@@ -0,0 +1,25 @@
+import React from 'react'
+
+export default class Poll extends React.Component {
+  constructor (props) {
+    super(props)
+    if (typeof window !== 'undefined') {
+      this._pollId = window.setInterval(this.props.function, this.props.interval)
+    }
+  }
+
+  componentWillUnmount () {
+    if (typeof window !== 'undefined') {
+      window.clearInterval(this._pollId)
+    }
+  }
+
+  render () {
```

Comment:
Why accept a `function` prop and function as child? Unless we want to capture the return value of the `function` prop, and pass it to the child function, I think we could just render `children`?

Reply:
lol, made this late last night, just copied the child logic from another component.

Reply:
@staydecent thinking about this more... if we just render children then Props doesn't allow props to pass through it which kinda sucks! I'm not sold on doing HOCs as jsx I prefer calling them as functions with pipe, and requiring that they always pass along the props they're given. 

```
pipe(
  withState(...),
  poll(...)
)(Base)
```

Reply:
if we're sticking with `index.js` and `base.js`, I prefer to not put jsx inside index.js. IMO the reason we use jsx at all is because it's helpful for visualizing, but that doesn't really apply when the components can't be seen

Reply:
I actually ditched base as I found it annoying. I found myself putting too much logic in index, where with single component files I just break out my components more, or move reusable functions to util.

can chat with @coryschadt to see if neighbors has been more confusing without `base` files.

Passing props to children is up to the component implementation (via `cloneElement`).

Reply:
With no base file there's more logic in index.
With base you can still break components into smaller pieces 😕

Reply:
Most components just end up looking like this: https://github.com/inputlogic/neighbors-mobile/blob/master/components/screens/Messages/index.js

Splitting out into base doesn't have benefits for me.

JSX is more familiar than pipe for the greater react community. Using pipe is what lead to confusing component files, which I think base meant to solve. But, just using JSX, component files are easy to follow for general React developers.

Reply:
As an example, suppose that conversations there had a "load more" button. when it's clicked you want to show the conversations taht are already loaded and page two of the same endpoint. `{endpoint: 'conversations'}` and `{endpoint: 'conversations', page: 2}` and so on

Reply:
For the record, I'm totally cool with having a `WithRequest` wrapping component for situations where we don't need something more complicated. My issue is with enforcing that as the only way.

something like this:
```javascript
// withRequest is a hoc, WithRequest is a component version.
export const WithRequest = ({request, children}) => withRequest(request)(children)
```

This way we can still use `withRequest` to build other more complex components.

Reply:
Yeah that makes sense.

Could each of those search endpoints reuse a component like:

```
<SearchResults resource='organizations' />
<SearchResults resource='users' />
<SearchResults resource='events />
```

Since they are independent requests anyway, do they need to be aware? Could store result counts in global state to determine if you need to display "no results" message.

Or use a single search API endpoint: https://github.com/inputlogic/client-unplug-api/blob/master/project/views.py#L89

Currently WithRequest can be extends via `class` or used in JSX, not sure exactly what changes would be needed to support functional/piping.

Reply:
They do need to be aware because I only want one loading indicator based on if any of them haven't resolved. I also need an empty state if none of them match. 
The specifics of this scenario aren't that important though, just an example of how limiting withrequest to a jsx parent component makes it less useful.

not storing requests in global state is another issue entirely, I don't see the benefit of only storing request results in localstate, especially if you have more than one component making the same request. I could have 10 components that need the `me` api result on the same page at any given time.

Reply:
Those issues can be solved by storing the loading state for each SearchResult globally.

The state for WithRequest is still global, just not redux/store. It's a shared single CACHE object. Can be updated to use redux/store with minimal changes.